Capture basic show-tech from switches via console

pau-hedgehog opened the pull request.

Frostman commented:
@pau-hedgehog why replace ssh with console?

Frostman commented on pkg/hhfab/vlabhelpers.go:
```diff
@@ -582,6 +628,40 @@ func (c *Config) VLABShowTech(ctx context.Context, vlab *VLAB) error {
+	var serverConsoleCleanup func()
+	if _, err := exec.LookPath(VLABCmdExpect); err == nil {
+		if cleanup, path, err := c.prepareShowTechConsoleScript(); err == nil {
+			switchConsoleCleanup = cleanup
```
you can just defer cleanup here (nit pick, doesn't have to fix)

Frostman commented on pkg/hhfab/vlabhelpers.go:
```diff
@@ -582,6 +628,40 @@ func (c *Config) VLABShowTech(ctx context.Context, vlab *VLAB) error {
+			slog.Warn("Failed to prepare console show-tech script; console fallback disabled", "err", err)
+		}
+		if cleanup, path, err := c.prepareShowTechServerConsoleScript(); err == nil {
+			serverConsoleCleanup = cleanup
```
same here (nit pick, doesn't have to fix)

Frostman commented on cmd/hhfab/main.go:
```diff
@@ -423,6 +423,19 @@ func Run(ctx context.Context) error {
+		&cli.StringFlag{
+			Name:    "switch-username",
+			Usage:   "switch username for console show-tech",
+			EnvVars: []string{hhfab.VLABEnvSwitchUser, "HHFAB_SWITCH_USERNAME"},
```
One option for env variable is enough

pau-hedgehog commented:
> Need to get rid of the secret inherit and few minor comments, let me know when you're ready, I'll add secrets into runners

Thanks for looking into this. I've addressed the comments. Let's re-trigger when runners get the secrets

pau-hedgehog commented:
@Frostman this is ready to add secrets into runners so we can merge and hopefully start getting more insights on the failures
